src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -605,6 +631,36 @@ protected override void CreateIndicesForNonTypeMembers(ITypeDefinition typeDef)
             implementingMethods.Free();
         }
 
+        private void EmitParametersFromOriginalMetadata(IMethodDefinition methodDef, MethodDefinitionHandle handle)
+        {
+            var def = _previousGeneration.OriginalMetadata.MetadataReader.GetMethodDefinition(handle);
+
+            var parameters = def.GetParameters();
+            var paramDefinitions = this.GetParametersToEmit(methodDef);
+            int i = 0;
+            foreach (var param in parameters)
+            {
+                var paramDef = paramDefinitions[i];
+                _parameterDefs.AddUpdated(paramDef);
+                _existingParameterDefs.Add(paramDef, MetadataTokens.GetRowNumber(param));
+                _parameterDefList.Add(paramDef, methodDef);
+                i++;
+            }
+        }
+
+        private void EmitParametersFromDelta(IMethodDefinition methodDef, MethodDefinitionHandle handle)
+        {
+            var ok = _previousGeneration.FirstParamRowMap.TryGetValue(handle, out var firstRowId);
```

Comment:
It feels like it might be reasonable to fail in retail builds here if the handle isn't present, e.g. by changing the code to `var firstRowId = _previousGeneration.FirstParamRowMap[handle]`. However, I don't know exactly what the failure mode is here if we proceed into the loop with a `firstRowId` which is just `0` instead of being something meaningful for the given handle.

Reply:
This would still fail later, when trying to emit an EncLog row for Param row 0, and I think that would give a more descriptive exception that "element not found".

Realistically this can only fail if someone changes how EnC works in a way that would break a lot of things, like perhaps allowing direct creation of a delta without a baseline, so a debug assert feels appropriate to me.